pkg/test/ginkgo/options_monitor_events.go

```diff
@@ -155,6 +155,24 @@ func (o *MonitorEventsOptions) End(ctx context.Context, restConfig *rest.Config,
 	return nil
 }
 
+// AppendSyntheticTestIntervals is similar to End, except it takes list of new EventIntervals and appends
+// them to the current set of recordedEvents. Since these new EventIntervals happened within the
```

Comment:
Synthetic tests should not be injecting new events when they run, if so the events being moved around would be changing depending on when a test ran, later synthetics would get different results and we'd be headed for very weird and hard to debug bugs. 

We need to get these processed and added when we stop monitoring, likely in End.